scipy/signal/_short_time_fft.py

```diff
@@ -0,0 +1,1666 @@
+"""Short-time Fourier Transform Module.
+
+Implementation Notes (as of 2022-12)
+------------------------------------
+* When the minimal version of Python is bumped to 3.9 then
+  ``@lru_cache(maxsize=None)`` can be replaced with ``@cache`` from functools,
+  which should be marginally faster.
+* MyPy Version 0.931 does not support decorated property methods. Hence,
+  applying ``@property`` to methods decorated with `@lru_cache(maxsize=None)``
+  causes a MyPy error. Hence, the caching in method `lower_border_end()` is
+  done by hand. MyPy Version 0.950 does not seem to have this limitation.
+* Since `NDArray` is only available for Numpy versions 1.21 and above, the
+  methods `stft()`, `stft_detrend()`, `istft()`, and  `_ifft_func()` need to
+  use `NDArray` array as a parameter instead of `NDArray[complex]`.
```

Comment:
Just a fly by comment reading these 😃 We bumped all the dependencies here, minimal Python is now 3.9, Mypy is 1.0 and NumPy is 1.21.6.

Reply:
Thanks @tupui. I updated my  dependencies—I am still not quite best friends with mypy yet 😐